### PR TITLE
Add `from_horizons` classmethod

### PR DESCRIPTION
**Pull request by shreyasbapat**
Addresses #313

### CODE REVIEW

File: src/poliastro/twobody/orbit.py [hunk 1]
```diff
@@ -235,6 +237,24 @@ def from_body_ephem(cls, body, epoch=None):
...
+        elif plane == Planes.EARTH_ECLIPTIC:
+            refplane = "ecliptic"
+        obj = Horizons(id=name, epochs=epoch.jd).elements(refplane=refplane)
+        a = obj["a"][0] * u.au
+        ecc = obj["e"][0] * u.one
+        inc = obj["incl"][0] * u.deg
+        raan = obj["Omega"][0] * u.deg
+        argp = obj["w"][0] * u.deg
+        nu = obj["nu"][0] * u.deg
+        ss = classical.ClassicalState(Sun, a * (1 - ecc ** 2), ecc, inc, raan, argp, nu)
```
Comment (Juanlu001) [inline src/poliastro/twobody/orbit.py]: `*State` classes should never be used. I will make them private and remove them from the documentation.

Comment (Juanlu001) [reply src/poliastro/twobody/orbit.py]: No, you should use `Orbit.from_classical`

Comment (shreyasbapat) [reply src/poliastro/twobody/orbit.py]: I am sorry for using that! I didn't know that I must not  use `*State`

Comment (shreyasbapat) [reply src/poliastro/twobody/orbit.py]: Oh! Yes, obviously. I am sorry again!

File: src/poliastro/tests/tests_twobody/test_orbit.py [hunk 2]
```diff
@@ -384,3 +384,38 @@ def test_orbit_propagate_retains_plane(value):
     expected_frame = orig_frame.replicate_without_data(obstime=final_ss.epoch)
 
     assert final_ss.frame.is_equivalent_frame(expected_frame)
+
+
+def test_from_horizons_raise_valueerror():
+    with pytest.raises(ValueError) as exep:
+        ss = Orbit.from_horizons(name="Dummy")
+    assert (
+        "ValueError: Unknown target (Dummy). Maybe try different id_type?"
```
Comment (shreyasbapat) [inline src/poliastro/tests/tests_twobody/test_orbit.py]: If I mock the call, would I be able to test this? @Juanlu001

**Comment by Juanlu001** [comment]
Thanks a lot @shreyasbapat! I think this is a great new feature that I'm eager to start using :) Do you want to rewrite the Tesla Roadster notebook accordingly?

**Comment by shreyasbapat** [comment]
Sure! I will do it today. :)

**Comment by shreyasbapat** [comment]
Sure. I will do so in an hour.

**Pull request merged**